wowchemy/assets/js/wowchemy-slides.js

```diff
@@ -0,0 +1,146 @@
+import * as params from "@params";
+
+import {fixMermaid} from './wowchemy-utils';
+
+// Enable core slide features.
+enabledPlugins = [
+  RevealMarkdown,
+  RevealHighlight,
+  RevealSearch,
+  RevealNotes,
+  RevealMath,
+  RevealZoom,
+];
+
+const isObject = function (o) {
+  return o === Object(o) && !isArray(o) && typeof o !== "function";
+};
+
+const isArray = function (a) {
+  return Array.isArray(a);
+};
+
+const toCamelCase = (s) => {
+  return s.replace(/([-_][a-z])/gi, ($1) => {
+    return $1.toUpperCase().replace("-", "").replace("_", "");
+  });
+};
+
+const keysToCamelCase = function (o) {
+  if (isObject(o)) {
+    const n = {};
+
+    Object.keys(o).forEach((k) => {
+      n[toCamelCase(k)] = keysToCamelCase(o[k]);
+    });
+
+    return n;
+  } else if (isArray(o)) {
+    return o.map((i) => {
+      return keysToCamelCase(i);
+    });
+  }
+
+  return o;
+};
+
+// reveal configurations can be included in front matter under slides.reveal
+var pluginOptions = {}
+if(typeof params.slides.reveal_options === "undefined"){
+  pluginOptions = params.slides.reveal_options;
+}
+
+
+pluginOptions = keysToCamel(pluginOptions);
+
+// Enable presentation menu by default.
+if (pluginOptions.menu_enabled === undefined) {
+  pluginOptions.menu_enabled = true;
+}
+
+// configure menu if enabled
+if (pluginOptions.menu_enabled) {
+  enabledPlugins.push(RevealMenu);
+
+  // make sure we have a menu configuration so we can set defaults
+  pluginOptions.menu = {}
+  if(typeof pluginOptions.menu === "undefined"){
+    pluginOptions = pluginOptions.menu;
+  }
+}
+
+pluginOptions["plugins"] = enabledPlugins;
+
+Reveal.initialize(pluginOptions);
+
+// Disable Mermaid by default.
+if (params.slides.diagram === undefined) {
+  params.slides.diagram = false;
+}
+
+// Configure Mermaid only if diagrams are enabled.
+if (params.slides.diagram) {
+  //mermaid options
+  // mermaid: front matter configuration can be used to set mermaid options
+  // You can also use directives (see mermaid documentation)
+  var mermaidOptions = {}
+  if(typeof params.slides.diagram_options === "undefined"){
+    mermaidOptions = params.slides.diagram_options;
```

Comment:
```
  if(typeof params.slides.diagram_options === "undefined"){
    mermaidOptions = params.slides.diagram_options;
```

What is the purpose of checking for undefined and then also setting the pluginOptions option to undefined? Did you mean not undefined?

Reply:
yet not defined .. was using a ternary operator and you requested to use if instead for readability I guess
copied the condition without modification